tools/bug-report/pkg/bugreport/bugreport.go

```diff
@@ -0,0 +1,253 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bugreport
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/operator/pkg/util"
+	"istio.io/istio/tools/bug-report/pkg/archive"
+	cluster2 "istio.io/istio/tools/bug-report/pkg/cluster"
+	"istio.io/istio/tools/bug-report/pkg/config"
+	"istio.io/istio/tools/bug-report/pkg/content"
+	"istio.io/istio/tools/bug-report/pkg/filter"
+	"istio.io/istio/tools/bug-report/pkg/kubeclient"
+	"istio.io/istio/tools/bug-report/pkg/kubectlcmd"
+	"istio.io/istio/tools/bug-report/pkg/processlog"
+	"istio.io/pkg/log"
+	"istio.io/pkg/version"
+)
+
+const (
+	bugReportDefaultMaxSizeMb = 500
+	bugReportDefaultTimeout   = 30 * time.Minute
+	bugReportDefaultTempDir   = "/tmp/bug-report"
+)
+
+var (
+	bugReportDefaultIstioNamespace = "istio-system"
+	bugReportDefaultInclude        = []string{""}
+	bugReportDefaultExclude        = []string{"kube-system,kube-public"}
+)
+
+// GetRootCmd returns the root of the cobra command-tree.
+func GetRootCmd(args []string) *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:          "bug-report",
+		Short:        "Cluster information and log capture support tool.",
+		SilenceUsage: true,
+		Long: "This command selectively captures cluster information and logs into an archive to help " +
+			"diagnose problems. It optionally uploads the archive to a GCS bucket.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runBugReportCommand(cmd)
+		},
+	}
+	rootCmd.SetArgs(args)
+	rootCmd.AddCommand(version.CobraCommand())
+	addFlags(rootCmd, gConfig)
+
+	return rootCmd
+}
+
+var (
+	// Logs, along with stats and importance metrics. Key is path (namespace/deployment/pod/cluster) which can be
+	// parsed with ParsePath.
+	logs       = make(map[string]string)
+	stats      = make(map[string]*processlog.Stats)
+	importance = make(map[string]int)
+	// Aggregated errors for all fetch operations.
+	gErrors util.Errors
+	lock    = sync.RWMutex{}
+)
+
+func runBugReportCommand(_ *cobra.Command) error {
+	// Default of "." doesn't work for K8s names.
+	util.PathSeparator = "/"
```

Comment:
override global seems suspicious

Reply:
i was using the external pkg for expediency but it really wasn't the right one anyway. i've moved a simplified path inside a local pkg.